src/test/unit/middlewares/validation.spec.js

```diff
@@ -0,0 +1,193 @@
+    })
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('Should return error when body is not present', () => {
+    req.body = null
+
+    try {
+      validationMiddleware({})(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(1)
+    }
+  })
+
+  it('Should return error when required field is not present', () => {
+    validateRequired.mockImplementation(() => {
+      throw 2
+    })
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(2)
+    }
+    expect(validateRequired).toHaveBeenCalledWith('test1', mockSchema.test1.required, undefined)
+  })
+
+  it('Should call next when empty field is not required', () => {
+    req.body = {
+      test1: undefined
+    }
+
+    try {
+      validationMiddleware({
+        test1: {
+          required: false
+        }
+      })(req, res, next)
+    } catch (err) {
+      expect(err).not.toBeDefined()
+    }
+
+    expect(next).toHaveBeenCalled()
+  })
+
+  it('Should return error when field has wrong type', () => {
+    req.body = {
+      test1: 1
+    }
+
+    validateFunc.type.mockImplementation(() => {
+      throw 3
+    })
+
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(3)
+    }
+
+    expect(validateFunc.type).toHaveBeenCalledWith('test1', mockSchema.test1.type, req.body.test1)
+  })
+
+  it('Should return error when field has wrong length', () => {
+    req.body = {
+      test1: '123'
+    }
+
+    validateFunc.length.mockImplementation(() => {
+      throw 4
+    })
+
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(4)
+    }
+
+    expect(validateFunc.length).toHaveBeenCalledWith('test1', mockSchema.test1.length, req.body.test1)
+  })
+
+  it("Should return error when field doesn't match regex", () => {
+    req.body = {
+      test1: 'abcd123'
+    }
+
+    validateFunc.regex.mockImplementation(() => {
+      throw 5
+    })
+
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(5)
+    }
+
+    expect(validateFunc.regex).toHaveBeenCalledWith('test1', mockSchema.test1.regex, req.body.test1)
+  })
+
+  it("Should return error when field isn't enum value", () => {
+    req.body = {
+      test1: 'testtest'
+    }
+
+    validateFunc.enum.mockImplementation(() => {
+      throw 6
+    })
+
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(err).toEqual(6)
+    }
+
+    expect(validateFunc.enum).toHaveBeenCalledWith('test1', mockSchema.test1.enum, req.body.test1)
+  })
+
+  it('Should not call next function when error appear', () => {
+    validateRequired.mockImplementation(() => {
+      throw 2
+    })
+    try {
+      validationMiddleware(mockSchema)(req, res, next)
+    } catch (err) {
+      expect(next).not.toHaveBeenCalled()
+    }

```

Comment:
Check if error that was caught from `validateRequired` and not from another place. And try to check calling of `next` outside of `try-catch`.

Reply:
moved expect(next).not.toHaveBeenCalled() out catch block and added expect(err).toBeDefined() to catch block
```js
  it('Should not call next function when error appear', () => {
    validateRequired.mockImplementation(() => {
      throw 2
    })
    try {
      validationMiddleware(mockSchema)(req, res, next)
    } catch (err) {
      expect(err).toBeDefined()
    }
    expect(next).not.toHaveBeenCalled()
  })
```

Reply:
moved expect(next).not.toHaveBeenCalled() out from catch block and added expect(err).toBeDefined()
```js
  it('Should not call next function when error appear', () => {
    validateRequired.mockImplementation(() => {
      throw 2
    })
    try {
      validationMiddleware(mockSchema)(req, res, next)
    } catch (err) {
      expect(err).toBeDefined()
    }
    expect(next).not.toHaveBeenCalled()
  })
```